x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/data/BlockAccountingTests.java

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.compute.data;
+
+import org.apache.lucene.util.BytesRef;
+import org.elasticsearch.common.util.BigArrays;
+import org.elasticsearch.common.util.BytesRefArray;
+import org.elasticsearch.test.ESTestCase;
+import org.hamcrest.Matcher;
+
+import java.util.BitSet;
+
+import static org.apache.lucene.util.RamUsageEstimator.alignObjectSize;
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.greaterThanOrEqualTo;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.lessThanOrEqualTo;
+
+public class BlockAccountingTests extends ESTestCase {
+
+    // These value may differ on different platform. If so, simply increase to their maximum observed value.
+    // The point is to simply ensure that the bass size is "reasonably" small.
+    static final long BASE_ARRAY_VECTOR_SIZE = 40L;
+    static final long BASE_ARRAY_BLOCK_SIZE = 64L;
```

Comment:
I'm not trilled with these, but we just need to sort of reasonable upper bound. They can be adjusted as needed, if at all.

Reply:
@jpountz Oh! I didn't know about that. Will take a look.

Reply:
RamUsageTester - very nice!!! I used it in the tests and removed the hardcoded constants.